samples/subsys/usb/cdc_acm_composite/src/main.c

```diff
@@ -54,16 +54,20 @@ static void interrupt_handler(const struct device *dev, void *user_data)
 			struct ring_buf *rb = &peer->data->rb;
 
 			read = uart_fifo_read(dev, buf, sizeof(buf));
-			if (read) {
-				wrote = ring_buf_put(rb, buf, read);
-				if (wrote < read) {
-					LOG_ERR("Drop %zu bytes", read - wrote);
-				}
+			if (read < 0) {
+				LOG_ERR("Failed to read UART FIFO");
+				read = 0;
+			};
+
+			wrote = ring_buf_put(rb, buf, read);
```

Comment:
Why not do it like this instead:
```
if (read > 0) {
	wrote = ring_buf_put(rb, buf, read);
	...
} else if (read < 0) {
	LOG_ERR("Failed to read UART FIFO");
}
```
What is the point in calling (even potentially) `ring_buf_put()` with `size` 0?

Reply:
Very unlikely read will be 0, and if it should be safe to call ring_buf_put().

Reply:
My point was just that calling `ring_buf_put()` when nothing was read is needless and could be easily avoided, and it even was avoided in the original code, so I'm not quite convinced it is good to change the implementation this way. But of course, in this particular case it is not that important (although people sometimes use snippets from samples as patterns to use in their applications), so I won't insist on changing this.